lib/features/thread/data/repository/thread_repository_impl.dart

```diff
@@ -388,18 +391,17 @@ class ThreadRepositoryImpl extends ThreadRepository {
   }
 
   @override
-  Future<List<EmailId>> emptySpamFolder(Session session, AccountId accountId, MailboxId spamMailboxId) {
-    return mapDataSource[DataSourceType.network]!.emptyMailboxFolder(
+  Future<List<EmailId>> emptySpamFolder(Session session, AccountId accountId, MailboxId spamMailboxId) async {
+    final listEmailIdDeleted = await mapDataSource[DataSourceType.network]!.emptyMailboxFolder(
       session,
       accountId,
-      spamMailboxId,
-      (listEmailIdDeleted) async {
-        await _updateEmailCache(
-          accountId,
-          session.username,
-          newDestroyed: listEmailIdDeleted
-        );
-      },
-    );
+      spamMailboxId);
+
+    await _updateEmailCache(
+      accountId,
+      session.username,
+      newDestroyed: listEmailIdDeleted);
```

Comment:
IMO, no need to `update Cache` this case, delete all cache emails belong to this mailbox is better

Reply:
And apply `step by step` operations in Hive Cache `update` instead of parallel operations
```dart
  Future<void> update(
    AccountId accountId,
    UserName userName, {
    List<Email>? updated,
    List<Email>? created,
    List<EmailId>? destroyed
  }) async {
    final emailCacheExist = await _emailCacheClient.isExistTable();
    if (emailCacheExist) {
      final updatedCacheEmails = updated?.toMapCache(accountId, userName) ?? {};
      final createdCacheEmails = created?.toMapCache(accountId, userName) ?? {};
      final destroyedCacheEmails = destroyed?.toCacheKeyList(accountId, userName) ?? [];

      await Future.wait([
        _emailCacheClient.updateMultipleItem(updatedCacheEmails),
        _emailCacheClient.insertMultipleItem(createdCacheEmails),
        _emailCacheClient.deleteMultipleItem(destroyedCacheEmails)
      ]);
    } else {
      final createdCacheEmails = created?.toMapCache(accountId, userName) ?? {};
      await _emailCacheClient.insertMultipleItem(createdCacheEmails);
    }
  }
```

Reply:
> IMO, no need to `update Cache` this case, delete all cache emails belong to this mailbox is better

Without the `updateCache` function, we perform concurrent `add/edit/delete` tasks depending on the passed prarams.

Reply:
> And apply `step by step` operations in Hive Cache `update` instead of parallel operations
> 
> ```dart
>   Future<void> update(
>     AccountId accountId,
>     UserName userName, {
>     List<Email>? updated,
>     List<Email>? created,
>     List<EmailId>? destroyed
>   }) async {
>     final emailCacheExist = await _emailCacheClient.isExistTable();
>     if (emailCacheExist) {
>       final updatedCacheEmails = updated?.toMapCache(accountId, userName) ?? {};
>       final createdCacheEmails = created?.toMapCache(accountId, userName) ?? {};
>       final destroyedCacheEmails = destroyed?.toCacheKeyList(accountId, userName) ?? [];
> 
>       await Future.wait([
>         _emailCacheClient.updateMultipleItem(updatedCacheEmails),
>         _emailCacheClient.insertMultipleItem(createdCacheEmails),
>         _emailCacheClient.deleteMultipleItem(destroyedCacheEmails)
>       ]);
>     } else {
>       final createdCacheEmails = created?.toMapCache(accountId, userName) ?? {};
>       await _emailCacheClient.insertMultipleItem(createdCacheEmails);
>     }
>   }
> ```

Done